pkg/backup/backup_compaction.go

```diff
@@ -116,6 +118,12 @@ func maybeStartCompactionJob(
 	}
 	startTS, endTS := chain[start].StartTime, chain[end-1].EndTime
 	log.Infof(ctx, "compacting backups from %s to %s", startTS, endTS)
+	testingKnobs := execCfg.BackupRestoreTestingKnobs
+	if testingKnobs != nil && testingKnobs.RunBeforeTriggeringCompaction != nil {
```

Comment:
responding to your slack message about the complicated inc backup success state verification: could you put the testing knob at the top of the compaction resumer instead in planning?